### ISSUE TITLE
How to acccess non-default event from the CKEditor component?

### ISSUE THREAD

**Issue by johnniepop** [ckeditor/ckeditor5-react#203]
Hello,

I need to control the data being pasted into the editor. I saw the answer in #199 and it seems to be a close enough variant. The problem is that it still is pure javascript and I can't figure how to access it from the CKEditor component. Yes, I can wrap it in a function, but to what prop should I link that function?
So can you, please, give me a hint to the right syntax for this, or point me to proper code examples?

Regards

**Comment by johnniepop** [comment]
OK, I got the right syntax (quite easy though) and the key is to define the listener into the **onInit** prop. Like this:
`<CKEditor
            id={id}
            editor={ClassicEditor}
            data={value}
            onInit={editor => {
               const documentView = editor.editing.view.document
               documentView.on('paste', (event, data) => {      // 'clipboardInput' in place of 'paste' here would work too
                  console.log('Event properties:', event /*.clipboardData.getData('text')*/)
               })
            }}
            onChange={(event, editor) => {
               const data = editor.getData()
               inputOnChange(data)
            }}
         />`
Now, with this state of the code, my problem is that I don't have the actual pasted data. The commented **clipboardData** property is nowhere to be found under the event's hierarchy (it is not within the mentioned 'data' either).
So, any ideas, please?

Regards

**Comment by FilipTokarski** [comment]
Hi, sorry for the late reply. To access clipboard data on paste, you can use code like this:
```js
onReady={ editor => {
        editor.plugins.get( 'Clipboard' ).on( 'inputTransformation', ( evt, data ) => {
            for( const element of data.content.getChildren() ) {
                console.log( element );
        }
    } );
} }
```
Then if you want, inside the for loop, you can do something with the pasted elements, based on condition. Example of preventing pasting images:
```js
editor.plugins.get( 'Clipboard' ).on( 'inputTransformation', ( evt, data ) => {
    for( const element of data.content.getChildren() ){
        if( element.name === 'img' ){
            evt.stop();
        }
    }
} );
```
Or for example use only span elements:
```js
editor.plugins.get( 'Clipboard' ).on( 'inputTransformation', ( evt, data ) => {
    for( const element of data.content.getChildren() ){
        if( element.name === 'span' ){
            console.log( element )
        }
    }
} );
```
Docs you might find helpful:
- [React integration component properties](https://ckeditor.com/docs/ckeditor5/latest/builds/guides/integration/frameworks/react.html#component-properties)
- [Clipboard](https://ckeditor.com/docs/ckeditor5/latest/framework/guides/deep-dive/clipboard.html)
- [Element](https://ckeditor.com/docs/ckeditor5/latest/api/module_engine_model_element-Element.html)

**Comment by johnniepop** [comment]
Hello @FilipTokarski and Happy New Year!

Thanks for the tips! I really got what I needed from your examples. With some tweaks. To be more clear on my use case - I had to deal with a large text chunks pasted in the aditor's area (for instance +5K lines). This is a problem because even if the component accomodates the text eventually, it makes the browser tab non responsive for a minute or so. So the most logical strategy seemed to grab the paste eent and truncate its size to a more manageable value. 

With that context in mynd I have some remarks about your examples.
- I tried to use the **onReady** prop as the means to catch the paste event - _unsuccessfully_. Only **onInit** worked for me.
- The _for loop_ is obviously the key to getting the data but there is a little twist here. It is a bit weird but the contents of the data coming with the paste event have a slightly different structure depending on if there is only one element in it or more. For instance if only one paragraph is pasted then the data is accessed through **element._textData**. For two or more elements it already have to be accessed with **element.getChild(0)._textData**. This is a bit weird and was a bit of an excercise to get it to work.

Anyway, your input was helpful and fixed my problem. 

Regards
